### PR TITLE
Add documentation on defining configurables

### PR DESCRIPTION
**Pull request by RukshiW**

### CODE REVIEW

**Comment by RukshiW** [comment]
Fixes https://github.com/wso2-enterprise/choreo/issues/13295

File: en/docs/references/define-configurable-variables.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    1. Sign in to the Choreo Console at https://console.choreo.dev.
+
+    2. Under **Create a New Component**, click **REST API** to start creating a REST API.
+   
+    3. In the **Create REST API** dialog, click **Start From Scratch**. Next, enter a unique name for the REST API and click **Create**.
+   
+        !!! Info
+            To learn how to create a REST API with more details, try the [Create Your First REST API tutorial](../tutorials/create-your-first-rest-api.md).
+
+    4. Once the **Components** page opens, click **Edit Code** to open the Web Editor.
```
Comment (Nashaath) [inline en/docs/references/define-configurable-variables.md:29]:
```suggestion
    4. Once the **Overview** page opens, click **Edit Code** to open the Web Editor.
```

File: en/docs/references/define-configurable-variables.md [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    7. Click **Save**.
+   
+        You have completed editing the REST API configuration.
+
+4. To view the Pro-Code view of the REST API, click the icon marked in the image below:
+
+    ![Open pro-code view](../assets/img/configurables/open-code-view.png){.cInlineImage-full}
+
+    You will see the configurable you added in the code view as `configurable string email = ?;`.
```
Comment (Nashaath) [inline en/docs/references/define-configurable-variables.md:77]: Should we go with the term code view in step 4 as well? We seem to be using the term "Pro-Code view" there.

File: en/docs/references/define-configurable-variables.md [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+5. Commit and push the REST API changes you made to GitHub so that they are also available in the Choreo Console. For instructions, see [Create Your First REST API - Step 1.5: Commit the REST API to GitHub](../tutorials/create-your-first-rest-api.md#step-15-commit-the-rest-api-to-github).
+
+6. Open the browser tab in which you are running the Choreo console and refresh it.
```
Comment (Nashaath) [inline en/docs/references/define-configurable-variables.md:81]:
```suggestion
6. Open the browser tab in which you are running the Choreo Console and refresh it.
```

File: en/docs/references/define-configurable-variables.md [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+7. Click **Deploy** for your REST API.
```
Comment (Nashaath) [inline en/docs/references/define-configurable-variables.md:83]: Should we go with "Click **Deploy** in the left navigation menu."

File: en/docs/references/define-configurable-variables.md [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+    4. Once the **Components** page opens, click **Edit Code** to open the Web Editor.
+
+2. Now let's define the email address you want to return as a configurable by following these sub-steps:
+
+    1. Click the first **+** icon on the page, and in the **Add Constructs** pane that appears, click **Configurable**.
+   
+        ![Add Configurable](../assets/img/configurables/add-configurable.png){.cInlineImage-threeQuarters}
+   
+        The **Configurable** pane opens, displaying the template for the configurable.
```
Comment (Nashaath) [inline en/docs/references/define-configurable-variables.md:37]: We are repeating this in the next step too. We might have to consider updating the next step accordingly.

File: en/docs/references/define-configurable-variables.md [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+8. To deploy the REST API, click **Config & Deploy**. 
+
+    The **Configure & Deploy** pane opens on the left of the page. The **email** field in this pane is the configurable you added to the REST API implementation. Choreo saves the value you enter here in an Azure key vault.
+
+9. In the **email** field, enter any email address.
+
+10. Click **Deploy**.
+
+     Once Choreo completes the deployment, the **Configurables** field indicates that the Rest API has a configured key.
```
Comment (Nashaath) [inline en/docs/references/define-configurable-variables.md:93]:
```suggestion
     Once Choreo completes the deployment, the **Configurables** field indicates that the REST API has a configured key.
```

File: en/docs/references/define-configurable-variables.md (hunk 5 above)
Comment (RukshiW) [reply en/docs/references/define-configurable-variables.md]: Removed repeated text.

File: en/docs/references/define-configurable-variables.md (hunk 2 above)
Comment (RukshiW) [reply en/docs/references/define-configurable-variables.md:75]: Changed to "code view".

File: en/docs/references/define-configurable-variables.md (hunk 4 above)
Comment (RukshiW) [reply en/docs/references/define-configurable-variables.md]: Done

File: en/docs/references/define-configurable-variables.md [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+Now you have tried out adding a basic configurable. Proceed to the subsections to learn how to add configurables with default values and define configurable variables as you develop your components.
+
+## Define a configurable with a default value
+
+When you add a configurable to your component in the Web Editor, you can enter a specific value as the configurable value instead of `?` as shown below.
+
+```
+configurable <VARIABLE_TYPE> <VARIABLE_NAME> = <DEFAULT_VALUE>
+```
+e.g., If you want to add a configurable named password of which the value should be in string format and with `secret` as the default value, you can add the configurable statement as follows:
```
Comment (Nashaath) [inline en/docs/references/define-configurable-variables.md:106]: shall we avoid saying "and with" in the sentence? Perhaps just keep either and or with.

**Review by Nashaath — CHANGES_REQUESTED** [review]
Added a few minor suggestions.

**Review by Nashaath — APPROVED** [review]

**Pull request merged**